libs/parallelism/execution/include/hpx/execution/algorithms/let_error.hpp

```diff
@@ -0,0 +1,284 @@
+                    std::add_lvalue_reference_t<E>>;
+                static_assert(hpx::execution::experimental::is_sender<
+                                  std::decay_t<type>>::value,
+                    "let_error expects the invocable sender factory to return "
+                    "a sender");
+            };
+
+            // Type of the potential senders returned from the sender factor F
+            template <template <typename...> class Variant>
+            using successor_sender_types = hpx::util::detail::unique_t<
+                hpx::util::detail::transform_t<predecessor_error_types<Variant>,
+                    successor_sender_types_helper>>;
+
+            // The workaround for clang is due to a parsing bug in clang < 11
+            // in CUDA mode (where >>> also has a different meaning in kernel
+            // launches).
+            template <template <typename...> class Tuple,
+                template <typename...> class Variant>
+            using value_types = hpx::util::detail::unique_concat_t<
+                predecessor_value_types<Tuple, Variant>,
+                hpx::util::detail::concat_pack_of_packs_t<hpx::util::detail::
+                        transform_t<successor_sender_types<Variant>,
+                            value_types<Tuple, Variant>::template apply
+#if defined(HPX_CLANG_VERSION) && HPX_CLANG_VERSION < 110000
+                            >
+                    //
+                    >>;
+#else
+                            >>>;
+#endif
+
+            template <template <typename...> class Variant>
+            using error_types =
+                hpx::util::detail::unique_t<hpx::util::detail::prepend_t<
+                    hpx::util::detail::concat_pack_of_packs_t<hpx::util::
+                            detail::transform_t<successor_sender_types<Variant>,
+                                error_types<Variant>::template apply>>,
+                    std::exception_ptr>>;
+
+            static constexpr bool sends_done = false;
+
+            template <typename R>
+            struct operation_state
+            {
+                struct let_error_predecessor_receiver
+                {
+                    std::decay_t<R> r;
+                    std::decay_t<F> f;
+                    operation_state& os;
+
+                    template <typename R_, typename F_>
+                    let_error_predecessor_receiver(
+                        R_&& r, F_&& f, operation_state& os)
+                      : r(std::forward<R_>(r))
+                      , f(std::forward<F_>(f))
+                      , os(os)
+                    {
+                    }
+
+                    struct start_visitor
+                    {
+                        void operator()(std::monostate) const
+                        {
+                            std::terminate();
+                        }
+
+                        template <typename OS_>
+                        void operator()(OS_& os) const
```

Comment:
You have disabled this function for `E == std::monostate` below (https://github.com/STEllAR-GROUP/hpx/pull/5274/files#diff-99bae62aa9f979fbaf1b4aab04e665f35655abd25d684a45a23b11ebfb31f760R142). Would there be a need to do the same here?

Reply:
It's probably best to do so. Good catch!